content/docs/user-guide/how-to/best-practices.md

```diff
@@ -0,0 +1,136 @@
+# Best Practices for DVC Projects
+
+Asking questions on data science collaboration to data scientists, engineers, or
+managers, we'll get a variety of answers. DVC provides a systematic approach
+towards managing and collaborating on data science projects. You can manage your
+projects with DVC more efficiently using the practices listed here:
+
+- Source code and data versioning
+
+  You can use DVC to avoid discrepancies between
+  [revisions](https://git-scm.com/docs/revisions) of source code and versions of
+  data files, when the data doesn't fit into a traditional repository. DVC
+  replaces all large data files, models, etc. with small
+  [metafiles](doc/user-guide/dvc-files-and-directories) (tracked by Git). These
+  files point to the original data, which you can access by checking out the
+  required `revision`.
+
+- Experiments
+
+  You can make use of Git branches to document progress of training different
+  types of models on your data files in the same project. Create a branch for
+  each of the models and then utilise DVC features while working on that branch.
```

Comment:
This would need much more info. One experiment per branch is just one of the experimentation approaches. Please browse issues about Experiments both in this repo and in the core repo and/or ask us via Discord for more info.

Reply:
Also needs more on-point solutions and less explanation (if we're sticking with Best Practices as a How-to).

Reply:
Removed this. This can actually be a single doc under How-to section. It can be implemented through branches, tag etc. Will take a lot space here with all explanation and steps.

Reply:
Hmmm... I think it's definitely one of the most important things to add in best practices. Although there is an issue about writing a separate doc about this indeed: #816 — but I think both can exist, and the section in Best Practices can just give an overview, and later link to the separate doc, when it exists... Please restore it for the new PR you'll be creating.

Reply:
Moved to https://github.com/iterative/dvc.org/pull/1748#pullrequestreview-483189340. Resolving here.